plugins/module_utils/foreman_helper.py

```diff
@@ -165,10 +165,19 @@ def __init__(self, argument_spec, **kwargs):
         self.state = 'undefined'
         self.entity_spec = {}
 
+        # State recording for changed and diff reporting
+        self._changed = False
```

Comment:
is there a benefit of having a private variable, a property and a setter, instead of just exposing the var as `self.changed`?

Reply:
Yes. Glad that you ask. The setter prevents anyone from resetting it. I think, that goes well with the logic, that if something in the module resulted in a change, nothing else should be able to state the opposite.
And the setter, as well as the property is exposed to the actual modules without a risk.

Reply:
fair enough